wazo_call_logd/plugins/support_center/service.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from datetime import datetime, timezone
+from dateutil.relativedelta import relativedelta
+
+from .exceptions import QueueNotFoundException, RangeTooLargeException
+
+
+class QueueStatisticsService(object):
+    def __init__(self, dao):
+        self._dao = dao
+
+    def _generate_interval(self, interval, from_, until):
+        time_deltas = {
+            'hour': relativedelta(hours=1),
+            'day': relativedelta(days=1),
+            'month': relativedelta(months=1),
+        }
+
+        if not until:
+            today = datetime.now(tz=timezone.utc)
+            until = datetime(today.year, today.month, today.day, tzinfo=timezone.utc) + relativedelta(days=1)
+
+        if not from_:
+            # NOTE(afournier): force from in schema instead
+            interval = None
+
+        time_delta = time_deltas.get(interval, 'hour')
+
+        if time_delta == time_deltas['hour']:
+            if from_ + relativedelta(months=1) < until:
+                raise RangeTooLargeException(
+                    details='Maximum of 1 month for interval by hour'
+                )
+        if interval:
+            current = from_
+            next_datetime = current + time_delta
+            while current < until:
+                yield current, next_datetime
+                current = next_datetime
+                next_datetime = current + time_delta
+                if next_datetime > until:
+                    next_datetime = until
+        else:
+            yield from_, until
+
+    def get(self, tenant_uuids, queue_id, **kwargs):
+        queue_stats = list()
+
+        interval = kwargs.pop('interval', None)
```

Comment:
`<nitpicking>`
If you're going to pop `kwargs` with a default value you might as well add if to the function definition.
```python
def get(self, tenant_uuids, queue_id, interval=None, from_=None, until=None, **kwargs):
```
`</nitpicking>`